api/webhooks/mixins.py

```diff
@@ -0,0 +1,31 @@
+from contextlib import suppress
+
+import requests
+from rest_framework.decorators import action
+from rest_framework.permissions import IsAuthenticated
+from rest_framework.response import Response
+
+from .models import AbstractBaseWebhookModel
+from .permissions import TriggerSampleWebhookPermission
+from .serializers import WebhookURLSerializer
+from .webhooks import trigger_sample_webhook
+
+
+class TriggerSampleWebhookMixin:
+    webhook_type = None
+
+    @action(
+        detail=False,
+        methods=["POST"],
+        url_path="test",
+        permission_classes=[IsAuthenticated, TriggerSampleWebhookPermission],
+    )
+    def trigger_sample_webhook(self, request, **kwargs):
+        serializer = WebhookURLSerializer(data=request.data)
+        serializer.is_valid(raise_exception=True)
+        webhook = AbstractBaseWebhookModel(url=serializer.validated_data["url"])
```

Comment:
Feels a bit wrong to instantiate an abstract model here. Can we grab the concrete model class instead based on the webhook type instead? Should be able to just define a property on the mixin, right? 

```python
@property
def webhook_model_class(self):
    return {
        WebhookType.ORGANISATION: OrganisationWebhook,
        WebhookType.ENVIRONMENT: EnvironmentWebhook
    }[self.webhook_type]
```

Reply:
But in this case if we create another subclass we have to  modify the parent(this property method) which seems kinda odd as well?

Reply:
Or maybe define a dictionary(in webhook.py) that maps webhook type to model instance? But that results in a circular import on environment model

Reply:
Hmm... yes, that's a very good point. Perhaps just having the constant elsewhere then though would be sufficient. E.g. just below the definition of the `WebhookType` enum?

Reply:
Ugh, yeah, ok circular import is annoying. Can we just create a `webhooks.constants` module and put it there perhaps? Bit hacky again but I think it's better than the abstract model instantiation.

Reply:
ugh, we already do have `webhook.costants` module and the import graph is even uglier

Reply:
have moved this to a function.

Reply:
I've resolved the circular import by just moving the lifecycle hook to a signal instead.